### ISSUE TITLE
Missing epochs in metric data

### ISSUE THREAD

**Issue by eric-czech** [wandb/client#234]
I recently ran an experiment using WandbCallback where training of a keras model runs in two phases, each looping for 30 epochs.  Somehow though I only ended up with 58 steps recorded in the [run page ](https://app.wandb.ai/hammerlab/celldom/runs/zhkapsu3) (but there are definitely 60 in the logs and keras model histories).  

In general, are there fairly loose guarantees around persistence, or might be this have something to do with running two separate training loops?

**Comment by vanpelt** [comment]
Hey @eric-czech sorry for the delayed response, I'm only seeing the 58 steps recorded in our database.  The logging should be 100% persistent.  My guess is `wandb.log` isn't getting called for either the first or last epoch of each of your phases.  The logs indicate all 60 epochs occurred.  Would be willing to share the code you're using?
